### PR TITLE
feat: add append command

### PR DESCRIPTION
**Pull request by AlexStocks**
## Summary by CodeRabbit

* **New Features**
  * Added the APPEND command to append values to stored strings. The command creates new keys if they don't exist and returns the new string length. Properly handles type errors and enforces size limits.

* **Tests**
  * Added comprehensive test coverage for APPEND functionality, including edge cases and concurrent operations.

### CODE REVIEW

File: src/storage/src/redis_strings.rs [hunk 1]
```diff
@@ -98,6 +98,78 @@ impl Redis {
...
+        let _lock = ScopeRecordLock::new(self.lock_mgr.as_ref(), &key_str);
+
+        let string_key = BaseKey::new(key);
+        let encode_value = db
+            .get_opt(&string_key.encode()?, &self.read_options)
+            .context(RocksSnafu)?
+            .unwrap_or_else(Vec::new);
+
+        let new_value: Vec<u8>;
+        let mut ctime: u64 = Utc::now().timestamp_micros() as u64;
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:119]: 延迟计算 ctime，仅在“新建 或 过期覆盖”时需要创建ctime，避免不需要的系统调用

File: src/storage/src/redis_strings.rs [hunk 2]
```diff
@@ -98,6 +98,78 @@ impl Redis {
...
+
+        // Get lock for the key
+        let key_str = String::from_utf8_lossy(key).to_string();
+        let _lock = ScopeRecordLock::new(self.lock_mgr.as_ref(), &key_str);
+
+        let string_key = BaseKey::new(key);
+        let encode_value = db
+            .get_opt(&string_key.encode()?, &self.read_options)
+            .context(RocksSnafu)?
+            .unwrap_or_else(Vec::new);
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:116]: 这里使用 unwrap_or_default() 会有问题吗

File: src/storage/src/redis_strings.rs [hunk 3]
```diff
@@ -98,6 +98,78 @@ impl Redis {
...
+    pub fn append(&self, key: &[u8], value: &[u8]) -> Result<i32> {
+        let db = self.db.as_ref().context(OptionNoneSnafu {
+            message: "db is not initialized".to_string(),
+        })?;
+
+        // Get lock for the key
+        let key_str = String::from_utf8_lossy(key).to_string();
+        let _lock = ScopeRecordLock::new(self.lock_mgr.as_ref(), &key_str);
+
+        let string_key = BaseKey::new(key);
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:112]: 这里可以提前获取 encode 字节数组，后面 查询 db 和 写回db 时可以复用，减少一次编码
